### ISSUE TITLE
Accessibility: Narrator cannot focus entire texts and announce "Document selected" when pressing Ctrl+A to choose all texts in TextBox/ComboBox/DomainUpDown/NumericUpDown

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#9828]
### .NET version

.Net 8.0: 8.0.100-rc.2.23431.5

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, this issue cannot repro in .Net 6.0 and .Net 7.0 for TextBox/DomainUpDown/NumericUpDown, but repro in ComboBox

### Issue description

When using keyboard: Ctrl+A or mouse to choose all texts in TextBox/ComboBox/DomainUpDown/NumericUpDown, Narrator cannot focus entire texts and announce "Document selected" to user

This issue cannot repro in MaskedTextBox/RichTextBox

**The result in .Net 8.0 RC2 latest build:**
![image](https://github.com/dotnet/winforms/assets/26474449/7c340f96-19c7-4e48-bdcc-32524ad7ca93)
![SelectAll_8 0](https://github.com/dotnet/winforms/assets/26474449/29680e97-46da-4c28-9e55-0702722fce0c)

**The result in .Net 7.0:**
![image](https://github.com/dotnet/winforms/assets/26474449/ddedbe40-8585-4983-8aab-01ec61d30308)
![SelectAll_7 0](https://github.com/dotnet/winforms/assets/26474449/95017d39-af03-471c-8ade-f17a54c10835)

### Steps to reproduce

1. Create a winforms .Net Core application with targeting .Net 8.0
2. Add these controls: TextBox/ComboBox/DomainUpDown/NumericUpDown with texts
3. Build and run 
4. Use Narrator to focus on the individual text in these controls, then presse Ctrl+A to select all texts

or use the sample application:
[WinFormsApp16.zip](https://github.com/dotnet/winforms/files/12495019/WinFormsApp16.zip)

**Comment by merriemcgaw** [comment]
@Olina-Zhang can you see whether we have the same behavior in NVDA and/or JAWS?

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang can you see whether we have the same behavior in NVDA and/or JAWS?

Tested it in .Net 8 and .Net 7 with NVDA, they are same: NVDA's blue focus rectangle keeps on entire control, cannot focus on specific text, and entire text when pressing Ctrl+A to select all. But all announcement in NVDA looks good.
![NVDA](https://github.com/dotnet/winforms/assets/26474449/8f204ac9-e248-4399-a7b6-5db9df81fc24)

**Comment by Philip-Wang01** [comment]
Verified this issue in the latest .Net 8.0 SDK build: 8.0.100-rtm.23476.4, When using keyboard: Ctrl+A or mouse to choose all texts in TextBox/DomainUpDown/NumericUpDown, Narrator can focus entire texts and announce "Document selected" to user.
![test9828](https://github.com/dotnet/winforms/assets/86937911/6beadd6f-165f-4b4d-a3cc-9c4a523b2cb6)

But for Combobox control, this issue still repro. This issue also repro's on .NET 6.0 and 7.0, but cannot repro on .NET framework project. @Tanya-Solyanik @SimonZhao888 Do we need to open a new issue to track it?
![testcombobox](https://github.com/dotnet/winforms/assets/86937911/88c6ffa8-22a5-4e67-a822-ccef61b8b90c)

**Comment by Nora-Zhou01** [comment]
Verified this issue from main branch with winforms repo, The result is the same as .Net 8.0 SDK build: 8.0.100-rtm.23476.4.

**Comment by Tanya-Solyanik** [comment]
>But for Combobox control, this issue still repro.

@Philip-Wang01 - yes, please open a new issue.

**Comment by Olina-Zhang** [comment]
Verified with .NET SDK 8.0.100-rc.2.23477.19 test pass build, this issue was fixed for TextBox/DomainUpDown/NumericUpDown controls except ComboBox, with the same above testing result. For ComboBox, have a new issue: [10007](https://github.com/dotnet/winforms/issues/10007).
